src/scmdata/run.py

```diff
@@ -0,0 +1,1718 @@
+
+        Returns
+        -------
+        tuple
+        """
+        return (len(self._ts), len(self.time_points))
+
+    @property
+    def values(self) -> np.ndarray:
+        """
+        Timeseries values without metadata
+
+        Calls :func:`timeseries`
+        """
+        return np.asarray(self._ts)
+
+    @property
+    def meta(self) -> pd.DataFrame:
+        """
+        Metadata
+        """
+        return pd.DataFrame(
+            [ts.meta for ts in self._ts], index=[ts.name for ts in self._ts]
+        )
+
+    def _meta_column(self, col) -> pd.Series:
+        vals = []
+        for ts in self._ts:
+            try:
+                vals.append(ts.meta[col])
+            except KeyError:
+                vals.append(np.nan)
+
+        return pd.Series(vals, name=col, index=[ts.name for ts in self._ts])
+
+    def filter(
+        self,
+        keep: bool = True,
+        inplace: bool = False,
+        has_nan: bool = True,
+        **kwargs: Any,
+    ):
+        """
+        Return a filtered ScmRun (i.e., a subset of the data).
+
+        Note that this this does not copy the underlying time-series data so any modifications will be reflected in the caller
+        ``ScmRun``. This allows for the updating a subset of the timeseries directly.
+
+        ```
+        >>> df
+        <scmdata.ScmRun (timeseries: 3, timepoints: 3)>
+        Time:
+            Start: 2005-01-01T00:00:00
+            End: 2015-01-01T00:00:00
+        Meta:
+               model     scenario region             variable   unit climate_model
+            0  a_iam   a_scenario  World       Primary Energy  EJ/yr       a_model
+            1  a_iam   a_scenario  World  Primary Energy|Coal  EJ/yr       a_model
+            2  a_iam  a_scenario2  World       Primary Energy  EJ/yr       a_model
+        >>> df.filter(scenario="a_scenario")["extra_meta"] = "test"
+        >>> df
+        <scmdata.ScmRun (timeseries: 3, timepoints: 3)>
+        Time:
+            Start: 2005-01-01T00:00:00
+            End: 2015-01-01T00:00:00
+        Meta:
+               model     scenario region  ...   unit climate_model extra_meta
+            0  a_iam   a_scenario  World  ...  EJ/yr       a_model       test
+            1  a_iam   a_scenario  World  ...  EJ/yr       a_model       test
+            2  a_iam  a_scenario2  World  ...  EJ/yr       a_model        NaN
+
+            [3 rows x 7 columns]
+        ```
+
+        This functionality is different to how `scmdata.ScmDataFrame` works which always returns a copy. If you do not want to
+        change the parent `ScmRun` create a copy `ScmRun.copy`. Any changes to this copy will not be reflected in the parent.
+
+        Parameters
+        ----------
+        keep
+            If True, keep all timeseries satisfying the filters, otherwise drop all the
+            timeseries satisfying the filters
+
+        inplace
+            If True, do operation inplace and return None
+
+        has_nan
```

Comment:
can we get rid of this now? We don't have that problem anymore right?

Reply:
Not at the moment the filters are shared between ScmDataFrame and ScmRun. Maybe at a later point in time